external-resolver/src/test/scala/org/genivi/sota/resolver/test/random/Command.scala

```diff
@@ -173,12 +175,27 @@ object Command extends
     FilterGenerators.genFilter(s.packages.keys.toList, s.components.toList)
       .map(AddFilter(_))
 
+  private def genCommandAddComponent(s: RawStore): Gen[AddComponent] =
+    ComponentGenerators.genComponent.map(AddComponent(_))
+
   private def genCommandInstallPackage(s: RawStore): Gen[InstallPackage] =
     for {
       veh <- Store.pickVehicle.runA(s)
       pkg <- Store.pickPackage.runA(s)
     } yield InstallPackage(veh, pkg)
 
+  private def genCommandInstallComponent(s: RawStore): Gen[InstallComponent] =
+    for {
+      veh <- Store.pickVehicle.runA(s)
+      cmp <- Store.pickComponent.runA(s)
+    } yield InstallComponent(veh, cmp)
+
+  private def genCommandUninstallComponent(s: RawStore): Gen[UninstallComponent] =
+    for {
+      veh <- Store.pickVehicle.runA(s)
+      cmp <- Store.pickComponent.runA(s)
```

Comment:
For uninstall to succeed you probably need to pick a component that exists on the vehicle, rather than just any component.